### PR TITLE
Develop proper facial expressions handler – Stint 3

### PR DESCRIPTION
**Pull request by martinaxgloria**
This PR contains the improvements in the `ergoCubEmotions` module. In particular, the module now contains a service that tells the user which are the available emotions that can be set (it is simply called `availableEmotions()`), it is integrated in yarpmanager and some tests were done on a NVIDIA Xavier AGX on bench, displaying only static images.
Moreover, the module features the possibility to display videos but some tests have to be done yet.

### CODE REVIEW

File: src/modules/ergoCubEmotions/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## ergoCubEmotions 
+This repo contains the `ergoCubEmotions` modul that allows to display different facial expressions on ergoCub.
```
Comment (pattacini) [inline src/modules/ergoCubEmotions/README.md:2]:
```suggestion
This repo contains the `ergoCubEmotions` module that allows displaying several facial expressions on ergoCub.
```

File: README.md [hunk 2]
```diff
@@ -45,7 +45,6 @@ cmake -DERGOCUB_MODEL_GENERATE_SIMMECHANICS=BOOL:ON -DERGOCUB_MODEL_COPY_TO_SRC=
 make
 (make install)
 ```
-
```
Comment (pattacini) [inline README.md:48]: You could revert this change on the main README.

File: src/modules/ergoCubEmotions/ergoCubEmotions.h [hunk 3]
```diff
@@ -53,13 +38,13 @@ class ErgoCubEmotions : public yarp::os::RFModule, public ergoCubEmotions_IDL {
         double getPeriod();
 
         bool setEmotion(const std::string& command);
+        std::vector<std::string> availableEmotions();
```
Comment (pattacini) [inline src/modules/ergoCubEmotions/ergoCubEmotions.h:41]: Thrift can probably map the `list` as defined in the IDL file into `std::vector`.
At any rate, you may try to use `std::list` here, for better consistency.

**Review by pattacini — CHANGES_REQUESTED** [review]
Can't wait to see how videos will be rendered 😄

File: README.md (hunk 2 above)
Comment (martinaxgloria) [reply README.md:48]: Do you mean keeping the description of the module and its usage also in the main README?

Comment (pattacini) [reply README.md:48]: Nope, I've realized that the main README underwent a very minimal (formatting) change that is not wanted/necessary.

File: src/modules/ergoCubEmotions/ergoCubEmotions.h (hunk 3 above)
Comment (martinaxgloria) [reply src/modules/ergoCubEmotions/ergoCubEmotions.h:41]: Since thrift maps `list` as `std::vector`, if I try to use here `std::list` instead of vector, it returns an error when compiling.

File: README.md (hunk 2 above)
Comment (martinaxgloria) [reply README.md:48]: Ok, I reverted it.

File: src/modules/ergoCubEmotions/ergoCubEmotions.h (hunk 3 above)
Comment (pattacini) [reply src/modules/ergoCubEmotions/ergoCubEmotions.h:41]: Ok! Good to know.

**Review by pattacini — APPROVED** [review]

**Review by Nicogene — APPROVED** [review]

**Comment by Nicogene** [comment]
Merged, thanks!
